compiler/rustc_passes/src/dead.rs

```diff
@@ -577,7 +578,27 @@ impl DeadVisitor<'tcx> {
             self.tcx.struct_span_lint_hir(lint::builtin::DEAD_CODE, id, span, |lint| {
                 let def_id = self.tcx.hir().local_def_id(id);
                 let descr = self.tcx.def_kind(def_id).descr(def_id.to_def_id());
-                lint.build(&format!("{} is never {}: `{}`", descr, participle, name)).emit()
+
+                let prefixed = vec![(span, format!("_{}", name))];
+
+                let mut diag =
+                    lint.build(&format!("{} is never {}: `{}`", descr, participle, name));
+                diag.multipart_suggestion(
+                    "if this is intentional, prefix it with an underscore",
+                    prefixed,
+                    Applicability::MachineApplicable,
+                )
+                .note(&format!(
+                    "The leading underscore signals to the reader that while the {} may not be {}\n\
```

Comment:
This text is what I suggested, but I see now that it is too verbose. (And as sanjay noted to me in private message, the parenthetical example is somewhat awkward to show when the content in question isn't actually a field or variable.

Reply:
(@sunjay had asked at the outset about whether to add a parameter to dictate what kind of text to construct. I told them to eschew that option at the time, but I have now told them to go ahead and do it based on seeing this result.)